Content.Server/Administration/Managers/BanManager.cs

```diff
@@ -265,15 +302,72 @@ public async Task<string> PardonRoleBan(int banId, NetUserId? unbanningAdmin, Da
         return $"Pardoned ban with id {banId}";
     }
 
-    public HashSet<ProtoId<JobPrototype>>? GetJobBans(NetUserId playerUserId)
+    private HashSet<string> GetActiveRoleBans(NetUserId playerUserId, string banTypePrefix)
     {
         if (!_cachedRoleBans.TryGetValue(playerUserId, out var roleBans))
-            return null;
+            return new HashSet<string>();
+
+        var now = DateTime.UtcNow;
         return roleBans
-            .Where(ban => ban.Role.StartsWith(JobPrefix, StringComparison.Ordinal))
-            .Select(ban => new ProtoId<JobPrototype>(ban.Role[JobPrefix.Length..]))
+            .Where(ban => ban.Role.StartsWith(banTypePrefix, StringComparison.Ordinal) && (ban.ExpirationTime == null || ban.ExpirationTime > now))
+            .Select(ban => ban.Role[banTypePrefix.Length..])
             .ToHashSet();
     }
+
+    public HashSet<ProtoId<JobPrototype>> GetJobBans(NetUserId playerUserId)
+    {
+        var activeJobBans = GetActiveRoleBans(playerUserId, JobPrefix);
+        return activeJobBans.Select(role => new ProtoId<JobPrototype>(role)).ToHashSet();
+    }
+
+    public bool IsRoleBanned(NetUserId userId, IEnumerable<string> roles)
+    {
+        var roleBans = GetRoleBans(userId);
+
+        if (roleBans == null)
+            return false;
+
+        return roles.Any(role => roleBans.Contains(role));
+    }
+
+    #endregion
+
+    #region Antag Bans
+    public HashSet<ProtoId<AntagPrototype>> GetAntagBans(NetUserId playerUserId)
+    {
+        var activeAntagBans = GetActiveRoleBans(playerUserId, AntagPrefix);
+        return activeAntagBans.Select(role => new ProtoId<AntagPrototype>(role)).ToHashSet();
+    }
+
+    private bool IsBannedFromAntag(NetUserId userId, IEnumerable<string> antags)
+    {
+        var antagBans = GetAntagBans(userId);
+        var antagAllSelection = Loc.GetString("ban-panel-role-selection-antag-all-option");
+
+        if (antagBans == null)
+            return false;
+
+        if (antagBans.Contains(new ProtoId<AntagPrototype>(antagAllSelection)))
+            return true;
```

Comment:
This is going to make bans localisation dependent which is awful. Just check for "All" directly here and elsewhere as necessary.